incubator/metrics-http/src/main/java/io/aklivity/zilla/runtime/metrics/http/internal/HttpActiveRequestsMetricContext.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021-2022 Aklivity Inc
+ *
+ * Licensed under the Aklivity Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.metrics.http.internal;
+
+import static io.aklivity.zilla.runtime.engine.metrics.MetricContext.Direction.BOTH;
+import static io.aklivity.zilla.runtime.metrics.http.internal.HttpUtils.commonId;
+import static io.aklivity.zilla.runtime.metrics.http.internal.HttpUtils.streamDirection;
+
+import java.util.function.LongConsumer;
+
+import org.agrona.DirectBuffer;
+import org.agrona.collections.Long2LongHashMap;
+
+import io.aklivity.zilla.runtime.engine.binding.function.MessageConsumer;
+import io.aklivity.zilla.runtime.engine.metrics.Metric;
+import io.aklivity.zilla.runtime.engine.metrics.MetricContext;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.AbortFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.BeginFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.EndFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.FrameFW;
+import io.aklivity.zilla.runtime.metrics.http.internal.types.stream.ResetFW;
+
+public final class HttpActiveRequestsMetricContext implements MetricContext
+{
+    private final String group;
+    private final Metric.Kind kind;
+    private final FrameFW frameRO = new FrameFW();
+
+    public HttpActiveRequestsMetricContext(
+        String group,
+        Metric.Kind kind)
+    {
+        this.group = group;
+        this.kind = kind;
+    }
+
+    @Override
+    public String group()
+    {
+        return group;
+    }
+
+    @Override
+    public Metric.Kind kind()
+    {
+        return kind;
+    }
+
+    @Override
+    public Direction direction()
+    {
+        return BOTH;
+    }
+
+    @Override
+    public MessageConsumer supply(
+        LongConsumer recorder)
+    {
+        return new HttpActiveRequestsMetricHandler(recorder);
+    }
+
+    private final class HttpActiveRequestsMetricHandler implements MessageConsumer
+    {
+        private static final long INITIAL_STATUS = 0b11L;
```

Comment:
Let's use an Initial state of `0`, then mask in a bit for each direction as they close.

Reply:
done